src/functionality.py

```diff
@@ -1,35 +1,136 @@
+                          "rp_start_date", "rp_start_time", "rp_duration", "doc", "sign_up", "ongoing", "ended"]
+            writer = csv.DictWriter(file, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerow({"serial_code": code_creator(), "approved": "MODERATOR", "approved_id": 420, "local": 1, "rp_name": "Love Hotel is NOT REAL", "main_host": "Deez",
+                             "main_host_id": 0, "rp_start_date": 0, "rp_start_time": 10, "rp_duration": 5, "doc": "https://www.youtube.com/watch?v=YoU3r6ZK8xQ",
+                             "sign_up": 0, "ongoing": 0, "ended": 1})
+
+    return rp_list
+
+
+# This is to start the checks and if said file does not exist, will create one.
+ban_id_check()
+rp_id_check()
+
+
+class Functionality:
     def __init__(self, bot: commands.Bot = None, guild_details: Dict = None):
         @bot.event
         async def on_ready():
+            # This is to begin the task loop.
+            second_passing.start()
             print('Logged in as {0.user}'.format(bot))
 
+        # This is a task loop, where it will self-update every 10 minutes.
+        @tasks.loop(seconds=600.0)
+        async def second_passing():
+            print(time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(round(time.time()))))
+            update_rp_list()
+            update_ban_list()
+
+        # This will execute before the function <second_passing> will run.
+        @second_passing.before_loop
+        async def inform_check():
+            print('The Ten-Minute Self-Update is now Running.')
+
         @bot.event
         async def on_command_error(ctx, error):
             if isinstance(error, commands.CommandNotFound):
-                await ctx.send('Unrecognized command.')
+                await ctx.send('`Unrecognized command.`')
                 return
+
+            if isinstance(error, commands.MissingRequiredArgument):
+                await ctx.send("`Please input the required arguments.`")
+                return
+
             raise error
 
         @bot.event
         async def on_message(message):
+            # Looks through the ban_id.csv
+            ban_id_ = ban_id_check()
+
             if message.author == bot.user:
                 return
 
             if message.channel.name in guild_details.relaying_channels:
+                # Right here, it will delete the message and notify the user who tried using the bot that they are banned.
+                # The ban_id_[1:] is to skip the example ID.
+                for user in ban_id_[1:]:
+                    bool_ = int(user["ended"])
+                    bool_ = bool(bool_)
+                    if message.author.id == user["discord_id"] and bool_ is False:
+                        await message.delete(message)
+                        return await message.author.send("**You cannot access the bot due to your ban.**")
```

Comment:
I think "use" rather than "access" should be better here.

Reply:
Understandable